sdk/Managed/src/Microsoft.WindowsAzure.MobileServices/Table/Sync/MobileServiceSyncContext.cs

```diff
@@ -303,6 +303,28 @@ public Task CancelAndUpdateItemAsync(MobileServiceTableOperationError error, JOb
             });
         }
 
+        public Task UpdateOperationAsync(MobileServiceTableOperationError error, JObject item)
+        {
+            string itemId = error.Item.Value<string>(MobileServiceSystemColumns.Id);
+            return this.ExecuteOperationSafeAsync(itemId, error.TableName, async () =>
+            {
+                await this.TryUpdateOperation(error, item);
+                if (error.OperationKind != MobileServiceTableOperationKind.Delete)
+                    await this.Store.UpsertAsync(error.TableName, item, fromServer: true);
+            });
+        }
+
+        private async Task TryUpdateOperation(MobileServiceTableOperationError error, JObject item)
+        {
+            if (!await this.opQueue.UpdateAsync(error.Id, error.OperationVersion, item))
```

Comment:
I didn't realize that error.id == operation.id, can we update error's help text to say that.  As is, it implies it unique string...